### PR TITLE
Added COSE conformance tests

### PR DESCRIPTION
**Pull request by patrickzheng200**
Signed-off-by: Patrick Zheng [patrickzheng@microsoft.com](mailto:patrickzheng@microsoft.com)

### CODE REVIEW

File: signature/cose/conformance_test.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+	// sign1.Output.CBORHex is a manually computed CBOR hex used as ground
+	// truth in the conformance test.
+	want := hexToBytes(sign1.Output.CBORHex)
+	if !bytes.Equal(want, got) {
+		t.Fatalf("unexpected output:\nwant: %x\n got: %x", want, got)
+	}
+
+	// Verify using the same envelope struct
+	_, _, err = env.Verify()
```
Comment (JeyJeyGao) [inline signature/cose/conformance_test.go:72]: Do we need to create a new envelope by calling ParseEnvelope() before Verify()?

Comment (patrickzheng200) [reply signature/cose/conformance_test.go:72]: Oh yeah, let me update the comment, the part you mentioned is covered in the testVerify() function

File: signature/cose/conformance_test.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+	certChain := make([]interface{}, len(certs))
+	for i, c := range certs {
+		certChain[i] = c.Raw
+	}
+	msg.Headers.Unprotected[cose.HeaderLabelX5Chain] = certChain
+	msg.Signature = env.base.Signature
+
+	newEnv := envelope{
+		base: &msg,
+	}
```
Comment (binbin-li) [inline signature/cose/conformance_test.go:114]: not a strong opinion, I feel we can extract it out to a new submethod like `createNewEnv`?

**Review by binbin-li — APPROVED** [review]
lgtm overall

File: signature/cose/conformance_test.go (hunk 2 above)
Comment (patrickzheng200) [reply signature/cose/conformance_test.go:114]: Sure. Will update the code

**Pull request merged**